BUILD

```diff
@@ -86,6 +88,18 @@ genrule(
     ]),
 )
 
+genrule(
+    name = "generate_dist_lockfile",
+    srcs = [
+        "MODULE.bazel",
+        "//third_party/googleapis:MODULE.bazel",
+        "//third_party/remoteapis:MODULE.bazel",
+    ],
+    outs = ["MODULE.bazel.lock.dist"],
+    cmd = "touch BUILD && $(location //src:bazel) --output_user_root=$$PWD/tmp_bazel_root query --check_direct_dependencies=error --lockfile_mode=update :all && mv MODULE.bazel.lock $@",
```

Comment:
Should we delete `tmp_bazel_root` in case the action isn't run sandboxed?

Reply:
Also, does this really not contain any module extension information? Since that information may not be reproducible due to us not declaring all inputs for it, if there is any, we should probably strip it.

Reply:
very sure this can't contain any module extension information since 1) I checked the generated lockfile for my test run and 2) there's no .bzl file in the genrule inputs, so no extension could have been run anyway.